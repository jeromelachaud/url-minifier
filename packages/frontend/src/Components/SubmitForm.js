import React, { Component } from 'react'
import { InputField } from './InputField'
import { Button } from './Button'
import { Loader } from './Loader'
import { ShortUrl } from './ShortUrl'
import { Service } from '../Service'

export default class SubmitForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isLoading: false,
      shortUrl: '',
      message: '',
      query: '',
    }

    this.onChangeQuery = this.onChangeQuery.bind(this)
    this.onSubmit = this.onSubmit.bind(this)
  }

  onChangeQuery(event) {
    this.setState({
      query: event.target.value,
    })
  }

  onSubmit(e) {
    e.preventDefault()
    this.setState({
      isLoading: true,
    })
    Service.minify({
      payload: this.state.query,
    }).then(res => {
      this.setState({
        isLoading: false,
        shortUrl: res.shortUrl,
        message: res.message,
      })
    })
  }

  render() {
    let responseElement
    if (this.state.isLoading) {
      responseElement = <Loader />
    } else if (this.state.shortUrl === '') {
      responseElement = null
    } else {
      responseElement = (
        <div>
          <ShortUrl
            message={this.state.message}
            link={this.state.shortUrl}/>
        </div>
      )
    }
    return (
      <form
        onSubmit={this.onSubmit}>
        <InputField
          type="url"
          id="url"
          htmlFor="url"
          labelText="Enter the URL to minify"
          placeholder="URL"
          ariaLabel="Enter the URL to minify"
          onChange={this.onChangeQuery}/>
        <Button
          text="Minify this ☝️"/>
        {responseElement}
      </form>
    )
  }
}
