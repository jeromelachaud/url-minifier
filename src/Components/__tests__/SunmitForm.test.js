import React from 'react'
import { shallow } from 'enzyme'
import SubmitForm from '../SubmitForm'

const tree = shallow(
  <SubmitForm/>
)

it('renders correctly', () => {
  expect(tree).toMatchSnapshot()
})

it('should have a default isLoading state set to false', () => {
  expect(tree.state().isLoading).toBe(false)
})

it('should have a default shortUrl state set to an empty string', () => {
  expect(tree.state().shortUrl).toBe('')
})

it('should have a default message state set to an empty string', () => {
  expect(tree.state().message).toBe('')
})

it('should have a default query state set to be an empty string', () => {
  expect(tree.state().query).toBe('')
})
