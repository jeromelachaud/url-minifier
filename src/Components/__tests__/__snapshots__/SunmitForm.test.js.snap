// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <Component
    ariaLabel="Enter the URL to minify"
    htmlFor="url"
    id="url"
    labelText="Enter the URL to minify"
    onChange={[Function]}
    placeholder="URL"
    type="url"
  />
  <Component
    text="Minify this ☝️"
  />
</form>
`;
